### PR TITLE
Use PartitionKeyInterceptor from the sdk

### LINKED ISSUE #406: [data-plane] Use the new PartitionKeyInterceptor from the sdk
**Problem**
Now we use internal code for the partitionkey extension, the sdk recently added an object `PartionKeyInterceptor` that transparently handles it https://github.com/cloudevents/sdk-java/pull/260

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for? Developer

**Exit Criteria**
Remove all our code to handle the partitionkey and replace it with PartitionKeyInterceptor

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve? 1

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

Fixes #406 

## Proposed Changes

Use PartitionKeyInterceptor from the sdk

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ private static CircuitBreakerOptions createCircuitBreakerOptions(final DataPlane
 
     // TODO create a single producer per bootstrap servers.
     producerConfigs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, resource.getBootstrapServers());
+    producerConfigs.put(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, "io.cloudevents.kafka.PartitionKeyExtensionInterceptor.class");
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java:156]: Can you use class directly, more than the name of the class? `PartitionKeyExtensionInterceptor.class`

File: data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java [hunk 2]
```diff
@@ -142,16 +142,12 @@ public void testUnorderedConsumer(final Vertx vertx, final VertxTestContext cont
     final var events = consumerRecords.stream()
       .map(ConsumerRecord::value)
       .toArray(CloudEvent[]::new);
-    final var partitionKeys = Arrays.stream(events)
-      .map(PartitionKey::extract)
-      .collect(Collectors.toList());
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java:147]: Why this removal?

Comment (arghya88) [reply data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java]: I removed this class `PartitionKey`. Should we put it back?

Comment (slinkydeveloper) [reply data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java]: Nono, it's ok that we removed it

Comment (slinkydeveloper) [reply data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java]: but you should keep the test as is

Comment (slinkydeveloper) [reply data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java]: This change is fine https://github.com/knative-sandbox/eventing-kafka-broker/pull/445/files#diff-80fb1392a05875d0f3d92f4180a8d9392364a1d274dfcf8ecca7b12e41f8fc8fR145 but maybe you need a null check?

File: data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java [hunk 3]
```diff
@@ -143,7 +143,12 @@ public void testUnorderedConsumer(final Vertx vertx, final VertxTestContext cont
       .map(ConsumerRecord::value)
       .toArray(CloudEvent[]::new);
     final var partitionKeys = Arrays.stream(events)
-      .map(PartitionKey::extract)
+      .map(e -> {
+        final var partitionKey = e.getExtension("partitionkey");
+        if (partitionKey == null) {
+          return null;
+        }
+        return partitionKey.toString();})
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/integration/UnorderedConsumerTest.java:151]: Fix indentation

**Comment by arghya88** [comment]
@slinkydeveloper any idea why the unit test is failing with below

```
dev.knative.eventing.kafka.broker.dispatcher.http.HttpConsumerVerticleFactoryTest.lambda$shouldNotThrowIllegalArgumentExceptionIfNotDLQ$1(HttpConsumerVerticleFactoryTest.java:127)
	at dev.knative.eventing.kafka.broker.dispatcher.http.HttpConsumerVerticleFactoryTest.shouldNotThrowIllegalArgumentExceptionIfNotDLQ(HttpConsumerVerticleFactoryTest.java:127)
Caused by: org.apache.kafka.common.KafkaException: Class class io.cloudevents.kafka.PartitionKeyExtensionInterceptor cannot be found
```

**Comment by pierDipi** [comment]
Try replacing toString -> getName when setting interceptors

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java [hunk 4]
```diff
@@ -153,6 +154,7 @@ private static CircuitBreakerOptions createCircuitBreakerOptions(final DataPlane
 
     // TODO create a single producer per bootstrap servers.
     producerConfigs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, resource.getBootstrapServers());
+    producerConfigs.put(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, PartitionKeyExtensionInterceptor.class.toString());
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java:157]: Here

Comment (arghya88) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java]: Yeah..that was it..Thanks

**Review by pierDipi — COMMENTED** [review]
Thank you! Removing code is always fantastic!

/lgtm
/assign @slinkydeveloper

**Pull request merged**